server/shared/src/main/scala/org/http4s/server/middleware/EntityLimiter.scala

```diff
@@ -21,11 +21,29 @@ package middleware
 import cats.ApplicativeThrow
 import cats.data.Kleisli
 import fs2._
+import org.http4s.headers.Connection
+import org.http4s.headers.`Content-Length`
 
 import scala.util.control.NoStackTrace
 
 object EntityLimiter {
-  final case class EntityTooLarge(limit: Long) extends Exception with NoStackTrace
+  final case class EntityTooLarge(limit: Long) extends Exception with NoStackTrace { self =>
+    def toMessageFailure: MessageFailure = new MessageFailure {
```

Comment:
In the 1.0 series, I'd like to address this more concisely by making `EntityTooLarge` straightforwardly extend `MessageFailure`. This change would eliminate the redundant conversion `EntityTooLarge` -> `MessageFailure` during error handling.

Reply:
> In the 1.0 series, I'd like to address this more concisely by making `EntityTooLarge` straightforwardly extend `MessageFailure`.

Why can't we do this on the 0.23 series?